UofS-Pulse-Binfo/analyzedphenotypes
# Phenotype image

This PR will implement a Tripal 3 field that will fetch trait image/photo uploaded in Upload data functionality of this module.

## Metadata

 - Issue  https://github.com/UofS-Pulse-Binfo/analyzedphenotypes/issues/95




Documentation:

## Description



## Dependencies



## Testing?



1. Add this field to Definition field in Trait content type.
2. Move the this field out of disabled fields list in Manage Display. Click Save.
3. The fieldgroup/Tripal Pane should contain this class 'ap-phenotype-image'. This will allow
the JS to account for other element when positioning left or right. To update class list, click cog icon of fieldgroup/tripal pane
![Screen Shot 2019-06-13 at 1 03 20 PM](https://user-images.githubusercontent.com/15472253/59460239-d99fe600-8ddb-11e9-91fc-e078ffa6d251.png) then update the Extra CSS classes and add ap-phenotype-image. Click save.
4. The field has its own settings. Clicking the cog will show options for stacking order of image as well as the position. Select configuration options and click save.
5. Load Tripal 3 page for a specific trait with image.

Examples: 
![Screen Shot 2019-06-13 at 1 07 53 PM (2)](https://user-images.githubusercontent.com/15472253/59460451-5c28a580-8ddc-11e9-88ea-e2030501aaa7.png)
Top to Bottom, Float Left

![Screen Shot 2019-06-13 at 1 09 37 PM (2)](https://user-images.githubusercontent.com/15472253/59460536-9003cb00-8ddc-11e9-9389-6bdb60c9c725.png)
Left to Right, Float Right

![Screen Shot 2019-06-13 at 1 10 59 PM (2)](https://user-images.githubusercontent.com/15472253/59460599-c17c9680-8ddc-11e9-9b1a-c4f9d051cbdf.png)
Left to Right, No position

NOTE: When set to position none, the order with which the field appears in the manage fields is the order it will show in the frontend. For example, If phenotype image field was listed before Definition field, It will show the Image above the Definition text.

Content, image and text-image ratio will adjust as window resizes.

This field assumes that you are adding this field to a fieldgroup that has another field that has element where the value/content was entered through Drupal Textarea field.
![Screen Shot 2019-06-13 at 1 23 29 PM (2)](https://user-images.githubusercontent.com/15472253/59461517-cb06fe00-8dde-11e9-82d0-a706f79e24a1.png)
